lib/tools/ui/widgets/image_picker_on_tap.dart

```diff
@@ -0,0 +1,51 @@
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+import 'package:image_picker/image_picker.dart';
+import 'package:myecl/tools/constants.dart';
+import 'package:myecl/tools/functions.dart';
+
+class ImagePickerOnTap extends StatelessWidget {
+  final ImagePicker picker;
+  final ValueNotifier<Uint8List?> imageBytesNotifier;
+  final ValueNotifier<Image?> imageNotifier;
+  final void Function(TypeMsg, String) displayToastWithContext;
+  final Widget child;
+  final int? imageQuality;
+
+  const ImagePickerOnTap({
+    super.key,
+    required this.picker,
+    required this.imageBytesNotifier,
+    required this.imageNotifier,
+    required this.displayToastWithContext,
+    required this.child,
+    this.imageQuality,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      onTap: () async {
+        final crossFile = await picker.pickImage(
```

Comment:
Why did you name the file `crossFile` ?
To me, `selectedFile` describes best what it is

Reply:
Because `pickImage` returns an XFile object which is not really a file (see https://pub.dev/documentation/cross_file/latest/cross_file/XFile-class.html).

Reply:
Ok, still don't get why `crossFile`but it matches the doc